Add KDBAI vector store

bu2kx opened the pull request:
Addition of KDBAI vector store (https://kdb.ai).

Dependencies: `kdbai_client` v0.1.2 Python package.

Sample notebook: `docs/docs/integrations/vectorstores/kdbai.ipynb`

Tag maintainer: @bu2kx
Twitter handle: @kxsystems

baskaryan commented on libs/langchain/langchain/vectorstores/kdbai.py:
```diff
@@ -0,0 +1,243 @@
+    def __init__(
+        self,
+        table: Any,
+        embedding: Union[Embeddings, Callable],
```
preference to just take Embeddings (any vecstores that take both do so for vestigial + backwards compatibility reasons)

bu2kx replied:
Thanks @baskaryan for the speedy review, I will proceed with the changes in the next couple hours.

bu2kx replied:
Should be good following new commit:  https://github.com/langchain-ai/langchain/pull/12797/commits/ed5834c9520482f3a88d7de8983f74847906e84d

baskaryan commented on libs/langchain/langchain/vectorstores/kdbai.py:
```diff
@@ -0,0 +1,243 @@
+from typing import Any, Callable, Iterable, List, Optional, Tuple, Union
+
+import numpy as np
+import pandas as pd
```
pandas is optional dependency so should be imported within functions that need it

baskaryan commented on libs/langchain/pyproject.toml:
```diff
@@ -138,6 +138,7 @@ anthropic = {version = "^0.3.11", optional = true}
 aiosqlite = {version = "^0.19.0", optional = true}
 rspace_client = {version = "^2.5.0", optional = true}
 upstash-redis = {version = "^0.15.0", optional = true}
+kdbai_client = {version = "^0.1.2", optional = true}
```
prefer to only include optional dependencies in pyproject if they're needed for unit testing (the all extras group is deprecated), so no need to add here

baskaryan reviewed (commented):
looks awesome, few comments! happy to make changes myself if you want to give me access to the fork

baskaryan commented on libs/langchain/pyproject.toml:
```diff
@@ -306,6 +306,7 @@ all = [
     "amadeus",
     "librosa",
     "python-arango",
+    "kdbai_client",
```
you'll want to delete here, too

bu2kx replied:
Oups, done.

bu2kx commented:
Dear @baskaryan, any change to get this PR merged in ? It has been passing lint check and tests locally last Friday (but the `langchain` community is adding more checks and tests faster than I can follow :-) )

hwchase17 reviewed (commented):
this should now be moved to langchain-community, as that is where all integrations have moved

bu2kx committed:
Rebase KDBAI vectorstore integration on top of LangChain v0.1.0.

bu2kx committed:
Add KDB.AI provider documentation.

bu2kx committed:
Improve sample notebook and documentation for KDB.AI vectorstore.

bu2kx committed:
Finalize sample notebook and documentation for KDB.AI vectorstore.

bu2kx committed:
Finalize sample notebook and documentation for KDB.AI vectorstore.

bu2kx committed:
Fix langchain_community unit tests following addition of KDBAI vectorstore.

bu2kx committed:
Fix imports to pass unit tests and linting following addition of KDBAI vectorstore.

bu2kx commented:
> this should now be moved to langchain-community, as that is where all integrations have moved

Dear @hwchase17 thanks for your feedback. 
We have rebased our integration on top of LangChain v0.1.0 relocated our vector store under `langchain_community`.
Unit tests and linting are passing successfully (locally).

bu2kx committed:
Fix linting of the sample notebook for KDBAI vectorstore.

bu2kx commented:
Dear @baskaryan , we updated our vectorstore integration to LangChain v0.1.0 and tried to pass all tests and linting checks locally. Could you give us another CI run ? Regards

bu2kx committed:
Merge branch 'master' into KDB.AI

bu2kx committed:
Fix failing unit test in libs/langchain following the addition of KDBAI vectorstore.

bu2kx commented:
Dear @hwchase17 and @baskaryan , we updated our vectorstore integration to LangChain v0.1.0, fixed a remaining unit test (`tests/unit_tests/vectorstores/test_public_api.py::test_public_api`),  and tried to pass all tests and linting checks locally. Could you give us another CI run ? Regards

PS: Would there be an easy way to run your CI checks under our Github account ? (so I could validate everything in advance)

bu2kx commented:
@baskaryan Could we get this merged in now that CI checks passed ? Regards.

baskaryan commented on libs/community/langchain_community/vectorstores/kdbai.py:
```diff
@@ -0,0 +1,238 @@
+import numpy as np
+
+try:
+    import pandas as pd
```
can we import this in the functions that use it, the way we import kdbai_client?

bu2kx replied:
I made the changes following your feedback. Thanks.

baskaryan reviewed (commented):
can we remove any code changes from `libs/langchain`? only old integrations are re-exported from there for backwards compatibility reasons, all new integrations only need to touch `libs/community`. the old `libs/langchain` exports will be removed in the next minor version release of `langchain`

bu2kx committed:
Remove changes from the libs/langchain/ directory.

bu2kx committed:
Merge branch 'master' into KDB.AI

bu2kx committed:
[KDBAI vectorstore] Load numpy and pandas in functions where they are used and fix a bug following LangChain v0.1.0 migration.

bu2kx committed:
[KDBAI vectorstore] Load numpy and pandas in functions where they are used and fix a bug following LangChain v0.1.0 migration.

baskaryan reviewed (approved):
lgtm, thank you @bu2kx!
